src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedTovStar.hpp

```diff
@@ -0,0 +1,234 @@
+ * the hydrodynamics not the magnetic fields.
+ *
+ * Superposes a poloidal magnetic field on top of a TOV solution where the
+ * vector potential has the form
+ *
+ * \f{align*}{
+ *  A_{\phi} = A_b \rho^2 \max(p-p_{\mathrm{cut}}, 0)^{n_s}
+ * \f}
+ *
+ * where \f$A_b\f$ controls the amplitude of the magnetic field,
+ * \f$\rho^2=x^2+y^2=r^2-z^2\f$ is the cylindrical radius,
+ * \f$n_s\f$ controls the degree of differentiability, and
+ * \f$p_{\mathrm{cut}}\f$ controls the pressure cutoff below which the magnetic
+ * field is zero.
+ *
+ * In Cartesian coordinates the vector potential is:
+ *
+ * \f{align*}{
+ *   A_x&=-\frac{y}{\varpi^2}A_\phi, \\
+ *   A_y&=\frac{x}{\varpi^2}A_\phi, \\
+ *   A_z&=0,
+ * \f}
+ *
+ * For the poloidal field this means
+ *
+ * \f{align*}{
+ *   A_x &= -y A_b\max(p-p_{\mathrm{cut}}, 0)^{n_s}, \\
+ *   A_y &= x A_b\max(p-p_{\mathrm{cut}}, 0)^{n_s}.
+ * \f}
+ *
+ * The magnetic field is computed from the vector potential as
+ *
+ * \f{align*}{
+ *   B^i&=n_a\epsilon^{aijk}\partial_jA_k \\
+ *      &=-\frac{1}{\sqrt{\gamma}}[ijk]\partial_j A_k,
+ * \f}
+ *
+ * where \f$[ijk]\f$ is the total anti-symmetric symbol. This means that
+ *
+ * \f{align*}{
+ *   B^x&=\frac{1}{\sqrt{\gamma}} (\partial_z A_y-\partial_y A_z), \\
+ *   B^y&=\frac{1}{\sqrt{\gamma}} (\partial_x A_z - \partial_z A_x), \\
+ *   B^z&=\frac{1}{\sqrt{\gamma}} (\partial_y A_x - \partial_x A_y).
+ * \f}
+ *
+ * Focusing on the region where the field is non-zero we have:
+ *
+ * \f{align*}{
+ *   \partial_x A_y
+ *   &= A_b(p-p_{\mathrm{cut}})^{n_s}+
+ *     \frac{x^2}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_r p \\
+ *   \partial_y A_x
+ *   &= -A_b(p-p_{\mathrm{cut}})^{n_s}-
+ *     \frac{y^2}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_r p \\
+ *   \partial_z A_x
+ *   &= -\frac{yz}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp \\
+ *   \partial_z A_y
+ *   &= \frac{xz}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp
+ * \f}
+ *
+ * The magnetic field is given by:
+ *
+ * \f{align*}{
+ *   B^x&=\frac{1}{\sqrt{\gamma}}\frac{xz}{r}
+ *        A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp \\
+ *   B^y&=\frac{1}{\sqrt{\gamma}}\frac{yz}{r}
+ *        A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp \\
+ *   B^z&=-\frac{A_b}{\sqrt{\gamma}}\left[
+ *        2(p-p_{\mathrm{cut}})^{n_s} \phantom{\frac{a}{b}}\right. \\
+ *      &\left.+\frac{x^2+y^2}{r}
+ *        n_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_r p
+ *        \right]
+ * \f}
+ *
+ * Taking the small-\f$r\f$ limit gives the \f$r=0\f$ magnetic field:
+ *
+ * \f{align*}{
+ *   B^x&=0, \\
+ *   B^y&=0, \\
+ *   B^z&=-\frac{A_b}{\sqrt{\gamma}}
+ *        2(p-p_{\mathrm{cut}})^{n_s}.
+ * \f}
+ *
+ * While the amplitude \f$A_b\f$ is specified in the code, it is more natural
+ * to work with the magnetic field strength, which is given by \f$\sqrt{b^2}\f$,
+ * and in GCS units is
+ *
+ * \f{align*}{
+ *  |B_{\mathrm{CGS}}|&= \frac{\sqrt{4 \pi b^2}}{1 \mathrm{M}_{\odot}}
```

Comment:
I get ` \frac{c^3}{G^{3/2} sqrt{\epsilon_0} M_\odot} = 8.35258 x 10^19 Gauss`. 
So I don't understand the solar mass  in the first fraction of the intermediate result or the factor of 10^4 Gauss

Reply:
I took this from Eq. (15) of https://arxiv.org/pdf/1405.2144.pdf

Reply:
ugh; it's complete nonsense